nextjs-13/src/app/api/v1/newsletter/send/route.ts

```diff
@@ -0,0 +1,54 @@
+import { StatusCodes } from 'http-status-codes';
+import { NextResponse, NextRequest } from 'next/server';
+
+import appHandler from 'lib/handler/app.handler';
+import getError from 'lib/request/getError';
+import getInvalidParamsError from 'lib/request/getInvalidParamsError';
+import { countNewsletters } from 'repositories/newsletter.repository';
+import { sendMailQueue } from 'workers/email.worker';
+
+const validateEmail = (email) => {
+  return String(email)
+    .toLowerCase()
+    .match(
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|.(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+    );
+};
+
+export async function POST(req: NextRequest) {
+  return appHandler(req, async (currentUser) => {
+    try {
+      const body = await req.json();
+      const email = body.email;
+
+      if (!validateEmail(email)) {
+        return getError('Invalid email');
+      }
+
+      const ids = [...new Set(body.ids)];
+      if (!ids.length) {
+        return getError('Invalid newsletters');
+      }
+
+      const newslettersCount = await countNewsletters(currentUser.id, ids);
+      const allIdsAreValid = newslettersCount === ids.length;
+      if (!allIdsAreValid) {
+        return getError('Invalid newsletters');
+      }
```

Comment:
So I'm having 3 validations here. I moved [1 validation](https://github.com/nimblehq/js-framework-benchmark/pull/121/commits/85e4e48beebbea1373eb05a2cf1a7fd99725e3b3#diff-51dc1719a4b4e593dd59c5173315a29eecd52aa3934b72161701b06185dd69e7L35-L39) into [the job](https://github.com/nimblehq/js-framework-benchmark/pull/121/commits/85e4e48beebbea1373eb05a2cf1a7fd99725e3b3#diff-4ddeb037537718795b666a209c40b103adb4056c177d96344ad164198862decbR12).

This route still have [2 validations](https://github.com/nimblehq/js-framework-benchmark/pull/121/files#diff-51dc1719a4b4e593dd59c5173315a29eecd52aa3934b72161701b06185dd69e7R25-R32), but since now it looks more concise, I think I should just leave these 2 here 👀